x/autopilot/keeper/redeemstake.go

```diff
@@ -0,0 +1,84 @@
+package keeper
+
+import (
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	"github.com/armon/go-metrics"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	transfertypes "github.com/cosmos/ibc-go/v5/modules/apps/transfer/types"
+	channeltypes "github.com/cosmos/ibc-go/v5/modules/core/04-channel/types"
+
+	"github.com/Stride-Labs/stride/v9/x/autopilot/types"
+	stakeibckeeper "github.com/Stride-Labs/stride/v9/x/stakeibc/keeper"
+	stakeibctypes "github.com/Stride-Labs/stride/v9/x/stakeibc/types"
+)
+
+func (k Keeper) TryRedeemStake(
+	ctx sdk.Context,
+	packet channeltypes.Packet,
+	newData transfertypes.FungibleTokenPacketData,
+	packetMetadata types.StakeibcPacketMetadata,
+) error {
+	params := k.GetParams(ctx)
+	if !params.StakeibcActive {
+		return fmt.Errorf("packet forwarding param is not active")
+	}
+
+	// In this case, we can't process a liquid staking transaction, because we're dealing IBC tokens from other chains
+	if !transfertypes.ReceiverChainIsSource(packet.GetSourcePort(), packet.GetSourceChannel(), newData.Denom) {
+		return fmt.Errorf("the ibc tokens are not supported for redeem stake")
+	}
+
+	voucherPrefix := transfertypes.GetDenomPrefix(packet.GetSourcePort(), packet.GetSourceChannel())
+	stAssetDenom := newData.Denom[len(voucherPrefix):]
+	if !stakeibctypes.IsStAssetDenom(stAssetDenom) {
+		return fmt.Errorf("not a liquid staking token")
+	}
+
+	hostZoneDenom := stakeibctypes.HostZoneDenomFromStAssetDenom(stAssetDenom)
+
+	amount, ok := sdk.NewIntFromString(newData.Amount)
+	if !ok {
+		return fmt.Errorf("not a parsable amount field")
+	}
+
+	// Note: newData.denom is ibc denom for st assets - e.g. ibc/xxx
+	var token = sdk.NewCoin(newData.Denom, amount)
```

Comment:
double-check if the packet data is validated, otherwise this line panics if the denom or amount are invalid